app/routers/calendar_grid.py

```diff
@@ -0,0 +1,136 @@
+import locale
+from datetime import date, datetime, timedelta
+from typing import Any, Generator, Iterator, List
+
+CALENDAR = calendar.Calendar(0)
+DISPLAY_BLOCK = 100
+
+locale.setlocale(locale.LC_TIME, ("en", "UTF-8"))
+
+
+class Week:
+    WEEK_DAYS = 7
+    DAYS_OF_THE_WEEK = calendar.day_name
+
+
+class Day:
+    def __init__(self, date: datetime):
+        self.date = date
+        self.sday = self.date.strftime("%A")
+        self.dailyevents = [("Daily Event", "More Information")]
+        self.events = [
+            ("6PM", "Meeting with Ode"),
+            ("7PM", "Meeting with Sagi")
+        ]
+        self.css = {
+            'div': 'day',
+            'date': 'day-number',
+            'daily_event': 'month-event',
+            'daily_event_front': 'daily front bg-warmyellow',
+            'daily_event_back': 'daily back t-darkblue bg-lightgray',
+            'event': 'event',
+        }
+
+    def __str__(self) -> str:
+        return self.date.strftime("%d")
+
+    def display(self) -> str:
+        """Returns day date inf the format of 00 MONTH 00"""
+        return self.date.strftime("%d %B %y").upper()
+
+    @classmethod
+    def is_weekend(cls, date: datetime) -> bool:
+        """Returns true if this day is represent a weekend."""
+        return date.strftime("%A") in Week.DAYS_OF_THE_WEEK[-2:]
+
+
+class DayWeekend(Day):
+    def __init__(self, date: datetime):
+        super().__init__(date)
+        self.css = {
+            'div': 'day ',
+            'date': 'day-number t-gray',
+            'daily_event': 'month-event',
+            'daily_event_front': 'daily front bg-warmyellow',
+            'daily_event_back': 'daily back t-darkblue bg-lightgray',
+            'event': 'event',
+        }
+
+
+class Today(Day):
+    def __init__(self, date: datetime):
+        super().__init__(date)
+        self.css = {
+            'div': 'day t-darkblue bg-yellow',
+            'date': 'day-number',
+            'daily_event': 'month-event',
+            'daily_event_front': 'daily front t-lightgray bg-darkblue',
+            'daily_event_back': 'daily back t-darkblue bg-lightgray',
+            'event': 'event',
+        }
+
+
+class FirstDayMonth(Day):
+    def __init__(self, date: datetime):
+        super().__init__(date)
+        self.css = {
+            'div': 'day t-darkblue bg-lightgray',
+            'date': 'day-number',
+            'daily_event': 'month-event',
+            'daily_event_front': 'daily front t-lightgray bg-red',
+            'daily_event_back': 'daily back t-darkblue bg-lightgray',
+            'event': 'event',
+        }
+
+    def __str__(self) -> str:
+        return self.date.strftime("%d %b %y").upper()
+
+
+def create_day(day: datetime) -> Day:
+    """Return the currect day object according to given date."""
+    if Day.is_weekend(day):
+        return DayWeekend(day)
+    if day == date.today():
+        return Today(day)
+    if int(day.day) == 1:
+        return FirstDayMonth(day)
```

Comment:
As far as I see, the difference of the first day of the month, is css only (i.e. view)
If that's true, maybe we don't need a different class for it?
For example, we can store a boolean flag for saying it's the first day (or not), and than handle it in the UI part?
What do you think?

Reply:
The idea is that each type of day will be able to get different theme but maybe in the further also different functionalities.
As times goes I guess it will be more clear and then in case we can merge them and create a Theme Class for the css.